include/fmt/ranges.h

```diff
@@ -414,29 +416,79 @@ struct range_formatter<
     if (it != end && *it == 'n') {
       set_brackets({}, {});
       ++it;
+    } else {
+      bool check_for_s = false;
+      if (it != end && *it == '?') {
+        ++it;
+        detail::maybe_set_debug_format(underlying_, true);
+        set_brackets({}, {});
+        check_for_s = true;
+        is_debug = true;
+      }
+      if (it != end && *it == 's') {
+        if (!std::is_same<T, Char>::value) {
+          report_error("invalid format specifier");
+        }
+        if (!is_debug) {
+          set_brackets(detail::string_literal<Char, '"'>{},
+                       detail::string_literal<Char, '"'>{});
+        }
+        check_for_s = false;
+        is_string_format = true;
+        ++it;
+      }
+      if (check_for_s) {
+        report_error("invalid format specifier");
+      }
     }
 
     if (it != end && *it != '}') {
-      if (*it != ':') report_error("invalid format specifier");
+      if (is_string_format || *it != ':')
+        report_error("invalid format specifier");
       ++it;
     } else {
-      detail::maybe_set_debug_format(underlying_, true);
+      if (!is_string_format) detail::maybe_set_debug_format(underlying_, true);
     }
 
     ctx.advance_to(it);
     return underlying_.parse(ctx);
   }
 
+  template <typename Output, typename Iter, typename IterEnd, typename U = T,
+            enable_if_t<std::is_same<U, Char>::value, bool> = true>
+  auto write_debug_string(Output& out, Iter& it, IterEnd& end) const -> Output {
+    auto buf = basic_memory_buffer<Char>();
+    for (; it != end; ++it) {
+      auto&& item = *it;
+      buf.push_back(item);
+    }
+    format_specs spec_str{};
+    spec_str.type = presentation_type::debug;
+    return detail::write<Char>(
+        out, basic_string_view<Char>(buf.data(), buf.size()), spec_str);
+  }
+  template <typename Output, typename Iter, typename IterEnd, typename U = T,
+            enable_if_t<!(std::is_same<U, Char>::value), bool> = true>
+  auto write_debug_string(Output& out, Iter& it, IterEnd& end) const -> Output {
+    detail::ignore_unused(it);
+    detail::ignore_unused(end);
```

Comment:
Let's remove `it` and `end` parameter names to make `ignore_unused` unnecessary.